backend/raw/raw.go

```diff
@@ -0,0 +1,127 @@
+package raw
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/diskfs/go-diskfs/backend"
+)
+
+type rawBackend struct {
+	storage  fs.File
+	readonly bool
+}
+
+// Create a backend.Storage from provided fs.File
+func New(f fs.File, isReadonly bool) backend.Storage {
+	return rawBackend{
+		storage:  f,
+		readonly: isReadonly,
+	}
+}
+
+// Create a backend.Storage from a path to a device
+// Should pass a path to a block device e.g. /dev/sda or a path to a file /tmp/foo.img
+// The provided device/file must exist at the time you call OpenFromPath()
+func OpenFromPath(pathName string, isReadonly bool) (backend.Storage, error) {
```

Comment:
I think you can just name this `readOnly`, which is fairly common. 

Actually, now that I think about it, do we want to just do `Open()` and have mode, paralleling `os.Open()`? That would give us the ability to have `Open()` even create the file. Or do we think this is simpler?

Reply:
> and have mode, paralleling os.Open()

no other mode bits except os.O_RDONLY are configurable by user here, so I opted to a boolean flag. How could we need all mode bits (as seen in os.Open()) here?

Reply:
> no other mode bits except os.O_RDONLY are configurable by user here

Don't you have the equivalent of `Create()`, which would be `Open(path, os.O_CREATE|os.O_RDWR)`?

Again, I myself am kind of torn on this. I know that logical OR of flags is the way Unix-like has done it for over 50 years, but I never really liked it either.

Reply:
So, let's keep the boolean flag here?